pkg/tools/otelcol-config/smoke_test.go

```diff
@@ -0,0 +1,70 @@
+//go:build !windows
+
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func createSkeleton(dir string) error {
+	if err := os.Mkdir(filepath.Join(dir, ConfDotD), 0700); err != nil {
+		return err
+	}
+	if err := os.Mkdir(filepath.Join(dir, ConfDotDAvailable), 0700); err != nil {
+		return err
+	}
+	if err := os.WriteFile(filepath.Join(dir, ConfDotDAvailable, getHostMetricsFilename()), []byte("hostmetrics"), 0600); err != nil {
+		return err
+	}
+	if err := os.WriteFile(filepath.Join(dir, ConfDotDAvailable, ephemeralYAML), []byte("ephemeral"), 0600); err != nil {
+		return err
+	}
+	return nil
+}
+
+func TestLocallyManagedSmoke(t *testing.T) {
+	// run main against a tempdir with all locally managed flags
+	configDir := t.TempDir()
+	if err := createSkeleton(configDir); err != nil {
+		t.Fatal(err)
+	}
+	flags := []string{
+		"otelcol-config",
+		"--config", configDir,
+		"--add-tag", "foo=bar",
+		"--add-tag", "bar=baz",
+		"--delete-tag", "bar",
+		"--set-installation-token", "abcdef",
+		"--enable-hostmetrics",
+		"--disable-hostmetrics",
+		"--enable-ephemeral",
+		"--disable-ephemeral",
+		"--set-api-url", "https://example.com",
+		"--write-kv", ".hello.world = \"yes\"",
```

Comment:
what is the `write-kv` flag for?

Reply:
write-kv is a developer-oriented flag that modifies `conf.d/00-otel-config-settings.yaml`. It's a generic feature so that we can add support for new configuration in e2e tests without requiring changes to otelcol-config.